### PR TITLE
Add If/Else to IRgen

### PR DESCRIPTION
**Pull request by Akash-Nayar**
Changes:
- Remove `IfElse` type
  - `If` now handles `if` with and without `else` clause
  - Achieved by parser parsing absence of `else` as empty list of `else_stmts`
- Added IRgen for `if/else`

### CODE REVIEW

File: lib/irgen.ml [hunk 1]
```diff
@@ -44,7 +44,17 @@ let translate (binds, sfuncs): L.llmodule =
...
   in
 
+  let function_decls : (L.llvalue * sfunc) StringMap.t =
+    let func_decl m func =
+      let name = func.sfname
+      and param_types = 
+        Array.of_list (List.map (fun (t,_) -> lltype_of_dtype t) func.sparams)
+            in let ftype = L.function_type (lltype_of_dtype func.srtype) param_types in
+            StringMap.add name (L.define_function name ftype mdl, func) m in
+          List.fold_left func_decl StringMap.empty sfuncs in
```
Comment (20joshuaz) [inline lib/irgen.ml:54]: I think this is the same as `all_funcs` right above

Comment (Akash-Nayar) [reply lib/irgen.ml:54]: oh oopsie

Comment (Akash-Nayar) [reply lib/irgen.ml]: removed lol

File: lib/irgen.ml [hunk 2]
```diff
@@ -45,6 +45,7 @@ let translate (binds, sfuncs): L.llmodule =
   in
 
   let build_func_body fn = 
+    let (the_func, _) = StringMap.find fn.sfname all_funcs in
```
Comment (20joshuaz) [inline lib/irgen.ml:48]: was going to make this change in my PR, nice!

**Review by 20joshuaz — APPROVED** [review]
You're so cute

**Comment by Akash-Nayar** [comment]
😊

File: lib/irgen.ml (hunk 2 above)
Comment (Akash-Nayar) [reply lib/irgen.ml:48]: yessirrrr

**Pull request merged**